crates/partition-store/src/partition_store.rs

```diff
@@ -1075,17 +1077,25 @@ pub(crate) trait StorageAccess {
         &mut self,
         key: K,
         value: &V,
+    ) -> Result<()> {
+        self.put_kv_storage_codec(
+            key,
+            &ProtobufStorageWrapper::<V::ProtobufType>(value.clone().into()),
```

Comment:
Unrelated to this PR: It seems wrong that we have to clone the value just to pass in the `ProtobufStorageWrapper` as a reference to `put_kv_storage_codec`. Shouldn't `ProtobufStorageWrapper` work with references?